src/adc.rs

```diff
@@ -241,6 +254,52 @@ impl VTemp {
     pub fn disable(&mut self, adc: &mut Adc) {
         adc.rb.ccr.modify(|_, w| w.tsen().clear_bit());
     }
+
+    pub fn is_enabled(&self, adc: &Adc) -> bool {
+        adc.rb.ccr.read().tsen().bit_is_set()
+    }
+
+    pub fn convert_temp(vtemp: u16, vdda: u16) -> i16 {
+        const VDD_CALIB: i32 = 3300_i32;
+        let vtemp30_cal = i32::from(unsafe { ptr::read(VTEMPCAL30) }) * 100;
+        let vtemp110_cal = i32::from(unsafe { ptr::read(VTEMPCAL110) }) * 100;
+
+        let mut temperature: i32 = (vtemp as i32) * 100;
+        temperature = (temperature * (vdda as i32) / VDD_CALIB) - vtemp30_cal;
+        temperature *= (110 - 30) * 100;
+        temperature /= vtemp110_cal - vtemp30_cal;
+        temperature += 3000;
+        temperature as i16
+    }
+
+    pub fn read_vtemp(adc: &mut Adc) -> i16 {
+        let mut vtemp = Self::new();
+        let vtemp_preenable = vtemp.is_enabled(&adc);
+
+        if !vtemp_preenable {
+            vtemp.enable(adc);
+
+            // Double read of vdda to allow sufficient startup time for the temp sensor
+            adc.read_vdda();
+        }
+        let vdda = adc.read_vdda();
+
+        adc.stash_cfg();
+
+        adc.set_align(AdcAlign::Right);
```

Comment:
Do you really need to duplicate this code? Couldn't you just put this in a non-public `get_raw` method (or something)?

Reply:
I could put in a reset to default settings function.

Reply:
Done